src/chfn.c

```diff
@@ -521,28 +527,32 @@ static void get_old_fields (const char *gecos)
 	 * Now get the full name. It is the first comma separated field in
 	 * the GECOS field.
 	 */
-	cp = copy_field (old_gecos, fflg ? NULL : fullnm, slop);
+	cp = copy_field (old_gecos, fflg ? NULL : fullnm, slop,
+			fflg ? 0 : countof(fullnm));
 
 	/*
 	 * Now get the room number. It is the next comma separated field,
 	 * if there is indeed one.
 	 */
 	if (NULL != cp) {
-		cp = copy_field (cp, rflg ? NULL : roomno, slop);
+		cp = copy_field (cp, rflg ? NULL : roomno, slop,
+				rflg ? 0 : countof(roomno));
 	}
```

Comment:
What happens if one of these doesn't copy due to missing space?  Should that be an error?  Should the program fail?